validator/sawtooth_validator/journal/publisher.py

```diff
@@ -490,8 +494,9 @@ def on_check_publish_block(self, force=False):
                         self._pending_batches:
                     self._build_candidate_block(self._chain_head)
 
-                if self._candidate_block and \
-                        (force or self._pending_batches) and \
+                if self._candidate_block and (
+                            force or
+                            self._candidate_block.has_pending_batches()) and \
```

Comment:
A more informative commit message would have made it easier to understand what the change was.